src/Utils.ps1

```diff
@@ -32,16 +32,16 @@ function Invoke-Utf8ConsoleCommand([ScriptBlock]$cmd) {
         # A native executable that writes to stderr AND has its stderr redirected will generate non-terminating
         # error records if the user has set $ErrorActionPreference to Stop. Override that value in this scope.
         $ErrorActionPreference = 'Continue'
-        if ($currentEncoding.IsSingleByte) {
-            try { [Console]::OutputEncoding = [Text.Encoding]::UTF8 } catch [System.IO.IOException] {}
+        try { 
+            [Console]::OutputEncoding = [Text.Encoding]::UTF8
+            & $cmd
+            [Console]::OutputEncoding = $currentEncoding
+        }
+        catch [System.IO.IOException] {
+            & $cmd
         }
```

Comment:
Seems like wrapping `& $cmd` in the `try` could result in reexecuting `$cmd` if it happens to `throw IOException`.

Is there a reason not to keep the previous pattern, just drop the `IsSingleByte` checks?
```suggestion
        try { [Console]::OutputEncoding = [Text.Encoding]::UTF8 } catch [System.IO.IOException] {}
        & $cmd
```

Reply:
> Is there a reason not to keep the previous pattern, just drop the IsSingleByte checks?

GBK and other Asian encoding aren't single byte, it will cause returing garbled until I switch [Console]::OutputEncoding to utf-8.

BTW, I use powershell 7.0.3, on windows 10 x64 2004, machine default encoding is gb2312 not utf8.

Reply:
Sorry, I meant to drop the `if ($currentEncoding.IsSingleByte)` but otherwise keep the previous structure, i.e.

```pwsh
        try { [Console]::OutputEncoding = [Text.Encoding]::UTF8 } catch [System.IO.IOException] {}
        & $cmd
    }
    finally {
        try { [Console]::OutputEncoding = $currentEncoding } catch [System.IO.IOException] {}
```

Specifically `finally` seems preferable to running `$cmd` in a `catch`.